corehq/sql_accessors/sql_templates/get_reverse_indexed_cases_3.sql

```diff
@@ -0,0 +1,33 @@
+DROP FUNCTION IF EXISTS get_reverse_indexed_cases(TEXT, TEXT[]);
+
+CREATE FUNCTION get_reverse_indexed_cases(
+    domain_name TEXT,
+    case_ids TEXT[],
+    case_types TEXT[] DEFAULT NULL,
+    p_closed BOOLEAN DEFAULT NULL
+) RETURNS SETOF form_processor_commcarecasesql AS $$
+DECLARE
+    query_expr      TEXT := '
+    SELECT case_table.* FROM form_processor_commcarecasesql as case_table
+        INNER JOIN form_processor_commcarecaseindexsql as case_index_table
+            ON ( case_table.case_id = case_index_table.case_id AND case_index_table.domain = $1 )
+            JOIN (SELECT UNNEST($2) AS referenced_id) AS cx USING (referenced_id)
+    WHERE
+        case_table.domain = $1
+        AND case_table.deleted = FALSE';
```

Comment:
I'm not crazy about sticking this big a query expression in a string, any idea if I can keep it in the body somehow?  Was mostly just trying to mimic the structure used in the other SQL functions.

Reply:
I agree that this  seems weird. It's possible that plpgsql doesn't have a way to add filters other than string concatenation though.

@snopoke is that why our other functions do that?

Reply:
aside: "plpgsql" is such a letter salad - "we use plpgsql to interact with the postgresql ordbms, which replaces our internal tool, asdfghjkl;"

Reply:
If you want to make the query dynamic you have to either to string concat like you're doing now or string formatting like this: https://github.com/dimagi/commcare-hq/blob/b927357119edb2f92fe02cd3257434586f06f14d/corehq/sql_accessors/sql_templates/get_forms_by_state.sql#L19